From: Takashi Iwai <tiwai@suse.de>
Patch-mainline: never
Subject: mgag200: Kms only for secure boot

Compatibility patch taken from SUSE:SLE-11-SP3:GA/drm.

Joey Lee:
Replaced CAP_COMPROMISE_KERNEL by secure_modules().

Acked-by: Michal Srb <msrb@suse.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>
---
 drivers/gpu/drm/mgag200/mgag200_drv.c |    8 ++++++++
 1 file changed, 8 insertions(+)

Index: linux-3.0-SLE11-SP4/drivers/gpu/drm/mgag200/mgag200_drv.c
===================================================================
--- linux-3.0-SLE11-SP4.orig/drivers/gpu/drm/mgag200/mgag200_drv.c
+++ linux-3.0-SLE11-SP4/drivers/gpu/drm/mgag200/mgag200_drv.c
@@ -122,6 +122,11 @@ static int __init mgag200_init(void)
 		return -EINVAL;
 #endif
 
+	if (!secure_modules() && mgag200_modeset == -1) {
+		printk(KERN_INFO "This driver is only used in secure boot mode as default\n");
+		return -EINVAL;
+	}
+
 	if (mgag200_modeset == 0)
 		return -EINVAL;
 	return drm_pci_init(&driver, &mgag200_pci_driver);
